src/cli/api/local_api.py

```diff
@@ -382,20 +526,266 @@ def get_results(self, path: Path, all_results: bool) -> List[ResultType]:
+                    missing_files = []
+                    for item in ['app_' + s + '.py' for s in roles]:
+                        if not (experiment_input_path / item).is_file():
+                            missing_files.append(item)
+
+                    if missing_files:
+                        error_dict['warning'].append(f"{experiment_input_path} is missing the files: {missing_files}")
+
+    # TODO:
+    #  - Update confluence sequence diagram with new function
+    #  - Roundsetmanager validate_asset() should be used in the future when multiple rounds are used.
+    def validate_experiment_json_files(self, path: Path, error_dict: ErrorDictType) -> None:
+        """
+        This function validates if experiment.json contains valid json and if it passes schema validation.
+
+        Args:
+            path: The location of the experiment
+            error_dict: Dictionary containing error and warning messages of the validations that failed
+        """
+
+        experiment_file_path = path / 'experiment.json'
+        experiment_input_path = path / 'input'
+
+        # Validate experiment.json
+        if os.path.isfile(experiment_file_path):
+            experiment_schema = Path(os.path.join(BASE_DIR, 'schema', 'experiments', 'experiment.json', ''))
+
+            json_valid, message = validate_json_file(experiment_file_path)
+            if json_valid:
+                schema_valid, ve = validate_json_schema(experiment_file_path, experiment_schema)
+                if not schema_valid:
+                    error_dict['error'].append(ve)
+
+                experiment_data = read_json_file(experiment_file_path)
+                experiment_network_slug = experiment_data['asset']['network']['slug']
+
+                # Check if the chosen network exists
+                if self._get_network_info(experiment_network_slug):
+                    # Validate experiment nodes
+                    self.validate_experiment_nodes(experiment_file_path, experiment_data, error_dict)
+                    # Validate experiment channels
+                    self.validate_experiment_channels(experiment_file_path, experiment_data, error_dict)
+                else:
+                    error_dict['warning'].append(f"In file {experiment_file_path}: network '{experiment_network_slug}' "
+                                                 f"does not exist")
+
+                self.validate_experiment_application(path, experiment_data, error_dict)
+
+        # Validate application.json, network.json and result.json
+        if os.path.exists(experiment_input_path):
+            app_schema_path = Path(os.path.join(BASE_DIR), 'schema', 'applications', '')
+            files_list = ["application.json", "network.json", "result.json"]
+
+            for file in files_list:
+                if os.path.isfile(experiment_input_path / file):
+                    json_valid, message = validate_json_file(experiment_input_path / file)
```

Comment:
this combination of checking json and then checking schema happens a lot. Now leading to duplicate code fragments. Maybe create a method for it (then also the one boolean I commented on earlier is solved).